### PR TITLE
feat: added dynamic search feature

### PR DESCRIPTION
**Pull request by ajaman190**
**Description**
This pull request addresses issue #114.

**Changes made**
- Added the query to fetch savedSearches and their counts.
- Made the Contact and Collection query variables dynamic using useState.
- Modified savedSearches screen functionality.
- Fixed failing test cases.

**Screenshots**

https://github.com/glific/mobile/assets/84832565/f80db809-4780-4ed6-9eab-d4aeb50a5363

**Closes Issue**
Closes: #114

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "@react-native-async-storage/async-storage": "1.17.11",
     "@react-native-community/datetimepicker": "6.7.3",
     "@react-native-community/slider": "^4.4.2",
-    "@react-native-picker/picker": "^2.4.10",
+    "@react-native-picker/picker": "2.4.8",
```
Comment (mdshamoon) [inline package.json:21]: Why are we downgrading the version here?

File: package.json [hunk 2]
```diff
@@ -31,6 +31,7 @@
     "expo-video-thumbnails": "^7.2.1",
     "graphql": "^15.8.0",
     "moment": "^2.29.4",
+    "npx": "^10.2.2",
```
Comment (mdshamoon) [inline package.json:34]: Do we need npx? This can be installed globally on system to use.

**Review by mdshamoon — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (ajaman190) [reply package.json:21]: Actually version 2.4.10 was not compatible with the phone input dependency.

Comment (ajaman190) [reply package.json:21]: was giving warning on every yarn start

File: package.json (hunk 2 above)
Comment (ajaman190) [reply package.json:34]: I don't know how it got added, I'm removing it now.

**Pull request merged**